lib/3scale/client.rb

```diff
@@ -52,24 +52,37 @@ class Client
     private_constant :EXTENSIONS_HEADER
 
     def initialize(options)
-      if options[:provider_key].nil? || options[:provider_key] =~ /^\s*$/
-        raise ArgumentError, 'missing :provider_key'
+      @provider_key = options[:provider_key]
+      @service_token = options[:service_token]
+      @use_service_tokens = options[:use_service_tokens]
```

Comment:
I see, that is good point. Because service token is bound to one service.

I guess then this could be decent migration plan. Offer the flag and leave it to false by default and in major release turn it on by default. Maybe some deprecation warning on use of provider_key would be in order.